refactor(eslint): switch to gewis eslint configs

SuperVK opened the pull request:
<!-- Provide a general summary of your changes in the title above. -->
Refactor to switch to https://github.com/GEWIS/js-configs (atleast for prettier and eslint, tsconfig can be done later)

This PR is only for the config that changed, #532  is for the files that had to be changed to conform to eslint, #534 is where the prettier is run

# Description
<!--
What do you want to achieve with this PR? Why did you write this code? What problem does this PR solve?
Describe your changes in detail and, if relevant, explain which choices you have made and why.
When making changes to the UI, make sure to include comparison screenshots!
-->

## Related issues/external references
<!--
Format issues on GitHub as `#XXX`. Tickets from support.gewis.nl can also be auto-linked by using
`ABC-YYMM-XXX`.
-->

## Types of changes
<!-- What types of changes does your code introduce? Remove all the items that do not apply: -->
- Style _(Change that do not affect the functionality of the code)_

SuperVK committed:
refactor(eslint): use GEWIS/js-config files

SuperVK committed:
chore(eslint): remove unused packages

SuperVK committed:
refactor(eslint): remove no used var from the translation files

SuperVK committed:
refactor(eslint): add static class names order

SuperVK committed:
refactor(prettier): add prettier config

SuperVK committed:
ci(eslint): add ci for linting and building

Gijsdeman commented on lib/common/eslint.config.js:
```diff
@@ -0,0 +1,10 @@
+import { defineConfig } from "eslint/config";
+// Very ironic, but without the extensions eslint does not work
+// eslint-disable-next-line import/extensions
+import root from "../../eslint.config.js";
```
See comment on `eslint.config.js` in root of project.

Gijsdeman commented on apps/dashboard/eslint.config.js:
```diff
@@ -0,0 +1,50 @@
+import * as typescriptParser from "@typescript-eslint/parser";
+// Very ironic, but without the extensions eslint does not work
+// eslint-disable-next-line import/extensions
+import root from "../../eslint.config.js";
```
See comment on `eslint.config.js` in root of project.

Gijsdeman commented on eslint.config.js:
Perhaps consider removing this file and moving imports to each workspace instead of importing this config. This will clean up some weird-looking imports upstream.

Rules that you think are missing from the config(s) but should be applied to both workspaces can just be added to https://github.com/GEWIS/js-configs/tree/main/configs/eslint-config-vue instead, to slowly create a more consistent linting experience across projects.

SuperVK committed:
ci(eslint): build libraries before linting

SuperVK committed:
ci(docker): do not build docker on PR

SuperVK committed:
refactor(eslint): remove root eslint config

SuperVK committed:
refactor(eslint): fix lint issues (#532)

* refactor(eslint): fix lint issues in common

# Conflicts:
#	eslint.config.js

* refactor(eslint): fix lint issues in point of sale

# Conflicts:
#	apps/point-of-sale/src/components/ProductComponent.vue

* refactor(eslint): fix lint issues in dashboard pt1

* refactor(eslint): fix lint issues in dashboard pt2

* refactor(eslint): class name ordered

* refactor(eslint): fix rebase mistake

* refactor(eslint): fix p component

* refactor(prettier): ran prettier fix (#534)

* refactor(prettier): ran prettier fix

* refactor(prettier): change CRLF to LF

* refactor(prettier): move ts-expect-errors with the prettier format
